### PR TITLE
Set up SSE (Server sent events) for event signup counts

### PR DESCRIPTION
**Pull request by jeffplays2005**
Created the getActiveEvents to fetch present events

### CODE REVIEW

**Comment by jeffplays2005** [comment]
Note that this could probably do better than using just a Promise<void> statement

image

Manual tests conducted:
image

(Note that the listener also correctly ends when the connection drops)

File: server/src/data-layer/services/EventService.ts [hunk 1]
```diff
@@ -25,6 +26,24 @@ class EventService {
     return result.data()
   }
 
+  /**
+   * Fetches all events that have a start date in the future.
+   * Note that "active" means any event that is still upcoming, or "ready" to signup.
```
Comment (bcho892) [inline server/src/data-layer/services/EventService.ts:31]: does that mean we need to check the `end_date` hasn't expired?

Comment (bcho892) [reply server/src/data-layer/services/EventService.ts:31]: that raises another question, we might need an extra `event_date` for when the event actually is

**Comment by bcho892** [comment]
also PR name

File: server/src/data-layer/services/EventService.ts (hunk 1 above)
Comment (jeffplays2005) [reply server/src/data-layer/services/EventService.ts:31]: > does that mean we need to check the `end_date` hasn't expired?

No, not necessarily. The `start_date` variable is supposedly ≥ `end_date` so essentially any `start_date` that is ≥ today would be a "scheduled" event. 

> that raises another question, we might need an extra `event_date` for when the event actually is

But couldn't events be lasting several days?

Comment (bcho892) [reply server/src/data-layer/services/EventService.ts:31]: I think my reasoning is that the signup for the event is different from the actual event date

Comment (jeffplays2005) [reply server/src/data-layer/services/EventService.ts:31]: > I think my reasoning is that the signup for the event is different from the actual event date

Ahhh right, I supposed that an event start_date & end_date was the actual planned dates for the signups and that they close reservations from the point of start_date

**Review by bcho892 — APPROVED** [review]

**Pull request merged**